src/main/kotlin/value/VFun.kt

```diff
@@ -35,20 +37,19 @@ data class VFun(
     }
 
     fun verbInvoke(c: Context, args: List<Value>): Value {
-        getVerb()?.also { verb ->
-            if ((args.size < argNames.size) || (args.size > argNames.size && args.size > 1))
-                fail(E_RANGE, "lambda wants ${argNames.size} args but got ${args.size}")
-
-            return verb.call(c, vThis, listOf(), entryPoint, withVars = buildMap {
-                vars.forEach { (name, v) -> put(name, v) }
-                argNames.forEachIndexed { n, name ->
-                    args.getOrNull(n)?.also { put(name, it) } ?: fail(E_RANGE, "arg $name not found")
-                }
-                if (argNames.isEmpty() && args.size == 1) put("it", args[0])
-            })
-
-        } ?: fail(VMException.Type.E_SYS, "missing verb for VFun ($traitID.$name)")
-        return VVoid
+        if ((args.size < argNames.size) || (args.size > argNames.size && args.size > 1))
```

Comment:
Is this the same as saying `args.size != argNames.size`?  Or are there situations where it's valid to have a different number of args than what you got?

Reply:
There is one situation, hence this awkward boolean -- if you pass one arg, and there are no named args, that's valid because the unnamed passed arg will be treated as 'it'.  I could have maybe handled this more elegantly.